### PR TITLE
OPHJOD-730: Update OsaamisSuosittelija to use new API

### PR DESCRIPTION
**Pull request by jhyoty**
## Description
OPHJOD-730: Update OsaamisSuosittelija to use new API

## Related JIRA ticket
https://jira.eduuni.fi/browse/OPHJOD-730

### CODE REVIEW

**Review by juhaniko — APPROVED** [review]

File: src/api/osaamiset.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { client } from './client';
+import { components } from './schema';
+
+type OsaaminenDto = components['schemas']['OsaaminenDto'];
+const cache = new Map<string, OsaaminenDto>();
```
Comment (jansu76) [inline src/api/osaamiset.ts:5]: Cache was decided to be split into a different ticket / PR

File: src/api/osaamiset.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+export const osaamiset = {
+  /**
+   * Combines objects referencing osaamiset with osaamiset.
+   *
+   * @param objects Collection of objects to combine with osaamiset.
+   * @param key Property accessor for the osaaminen uri.
+   * @param combiner Function to combine the object with the osaaminen.
+   * @param signal Abort signal.
+   * @returns An array of objects combined with osaamiset.
+   */
```
Comment (jansu76) [inline src/api/osaamiset.ts:18]: Would be good to explain here that `combine()` makes (one or multiple) API calls to fetch the data. Otherwise one might assume that it combines some data that already exists.

(would a different name make sense, `complete()` maybe?)

File: src/api/osaamiset.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            client.GET('/api/osaamiset', {
+              signal,
+              params: { query: { uri: chunk, sivu: 0, koko: chunk.length } },
+            }),
+          ),
+        )
+      )
+        .flatMap((result) => result.data?.sisalto ?? [])
+        .forEach((osaaminen) => cache.set(osaaminen.uri, osaaminen));
+    }
```
Comment (jansu76) [inline src/api/osaamiset.ts:61]: Does the chunking and combining results warrant a unit test? Any interesting corner cases?

File: src/api/osaamiset.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  combine: async <T, R>(
+    objects: Iterable<T> | undefined,
+    key: (o: T) => string,
+    combiner: (object: T, osaaminen: OsaaminenDto) => R,
+    signal?: AbortSignal,
+  ) => {
+    if (!objects) {
+      return [];
+    }
+    const chunks: string[][] = [];
```
Comment (jansu76) [inline src/api/osaamiset.ts:28]: This previous comment would be worth keeping to explain the motivation for an unfamiliar reader (unless already completely covered by the method-level comment)

`// the list of competences can be quite long, so we split the request into chunks to avoid exceeding the maximum URL length`

**Review by jansu76 — CHANGES_REQUESTED** [review]

**Review by sauanto — APPROVED** [review]

**Review by jansu76 — APPROVED** [review]
:+1:
